### PR TITLE
add chart style

### PR DESCRIPTION
**Pull request by ujhujuh**
add some chart style, such as linePlusBarChart, multiBarHorizontalChart, multiChart.

- **linePlusBarChart:**

e
f

- **multiBarHorizontalChart:**

c
d

- **multiChart:**

b
a

### CODE REVIEW

File: superset/assets/visualizations/treemap.css [hunk 1]
```diff
@@ -6,10 +6,10 @@ text {
   font-weight: bold;
 }
 
-rect {
+/*rect {
```
Comment (xrmx) [inline superset/assets/visualizations/treemap.css]: leftover?

File: superset/forms.py [hunk 2]
```diff
@@ -987,6 +987,116 @@ def __init__(self, viz):
...
+                "default": '',
+            }),
+            'y_axis_label2': (TextField, {
+                "label": _("Y2 Axis Label"),
+                "default": '',
+            }),
+            'y_domain1': (TextField, {
+                "label": _("Y1 domain"),
+                "default": '',
+                "description": "the range of y1, for example: 0,100, it means from 0 to 100"
```
Comment (xrmx) [inline superset/forms.py]: you can make the string translatable

File: superset/forms.py [hunk 3]
```diff
@@ -987,6 +987,116 @@ def __init__(self, viz):
...
+            }),
+            'y_domain2': (TextField, {
+                "label": _("Y2 domain"),
+                "default": '',
+                "description": "the range of y2, for example: 0,100, it means from 0 to 100"
```
Comment (xrmx) [inline superset/forms.py]: same here

File: superset/viz.py [hunk 4]
```diff
@@ -1343,8 +1344,13 @@ def query_obj(self):
         d['groupby'] = set(gb + cols)
         if len(d['groupby']) < len(gb) + len(cols):
             raise Exception("Can't have overlap between Series and Breakdowns")
-        if not self.metrics:
+        if not self.metrics and fd.get('viz_type') != 'multi':
```
Comment (xrmx) [inline superset/viz.py]: you can move this check as "elif not self.metrics" to the if below

File: superset/viz.py [hunk 5]
```diff
@@ -1354,7 +1360,7 @@ def get_df(self, query_obj=None):
         fd = self.form_data
 
         row = df.groupby(self.groupby).sum()[self.metrics[0]].copy()
-        row.sort_values(ascending=False, inplace=True)
+        # row.sort_values(ascending=False, inplace=True)
```
Comment (xrmx) [inline superset/viz.py]: why so?

File: superset/assets/visualizations/treemap.css (hunk 1 above)
Comment (ujhujuh) [inline superset/assets/visualizations/treemap.css]: if not delete,  the multiBarHorizontalChart can't display, like that:
g

File: superset/viz.py (hunk 5 above)
Comment (ujhujuh) [inline superset/viz.py]: @xrmx 
add this line:(Disorder)
a

remove this line:(Order)
b

**Comment by ascott** [comment]
@ujhujuh thanks for this PR. These a great additions, however we are about to launch V2 of the explore/slice page and will be deprecating V1. You will need to make these changes to the V2 code in a new PR. You can find the front end code for V2 here: https://github.com/ascott/superset/tree/master/superset/assets/javascripts/explorev2

and you can see the V2 code path in views.py here: https://github.com/airbnb/superset/blob/master/superset/views.py#L1497

**Comment by ujhujuh** [comment]
@ascott explorev2 controls some of the buttons and views above the slice interface, but the addition of the left-hand condition and the chart style is modified in viz.py and nvd3_vis.js,  I did not change things in explorev1 or explorev2, so how to make changes?

**Pull request closed without merging**